_tests/nested-struct-slice-no-key.hcl

```diff
@@ -1,7 +1,8 @@
-Widget {
-  Foo = "bar"
-}
-
-Widget {
-  Foo = "baz"
-}
+Widget = [
+  {
+    Foo = "bar"
+  },
+  {
+    Foo = "baz"
+  },
+]
```

Comment:
While I think the encoding makes sense for `nested-slices.hcl`, I'm not sure the output for this one does. It's been a while since I've written TF files, but I think the original pattern here is more idiomatic (for example, the [`filter`](https://www.terraform.io/docs/providers/aws/d/ami.html) property on AWS AMI resources is repeated).

Do you think it'd be possible to have the primitive lists (and lists of lists) behave as you have them, while the repeated Widget maintains this original form? Admittedly, it's been an awful long time since I've gotten my hands dirty in this codebase.

Reply:
Thanks for the quick review & comments 👍 

Completely agree about it being more idiomatic, but unfortunately there's inconsistencies between it being accepted within TF files and not within tfvars files for config. Just wrote up a quick example to show:
[https://gist.github.com/jleeh/0e41ade30af5ea6dd09fd7aa84cf10d3](url)

If I use the var file of:
```hcl
mapList = [
  {
    key = "key1"
    value = "value1"
  },
  {
    key = "key2"
    value = "value2"
  }
]
```
I correctly get the output of:
```hcl
rendered = [
    key1=value1,
    key2=value2
]
```

Whereas, if I use:
```hcl
mapList = {
  key = "key1"
  value = "value1"
}

mapList = {
  key = "key2"
  value = "value2"
}
```

I get this error back from Terraform:
`multiple map declarations not supported for variables`

Although with your example of filters, you can specify it as a list of maps as well as multiple declarations. So to me IMO, it should be encoding multiple maps by explicit lists, just because it works for all uses.

Reply:
Alright, I'm convinced. That's a real bummer though that the behavior is different for the var files (and is probably worth upstreaming, if it hasn't been already).